app/models/manageiq/providers/azure/cloud_manager/scanning/job.rb

```diff
@@ -0,0 +1,173 @@
+class ManageIQ::Providers::Azure::CloudManager::Scanning::Job < VmScan
+  # Make updates to default state machine to take into account snapshots
+  def load_transitions
+    super.tap do |transitions|
+      transitions.merge!(
+        :start_snapshot     => {'before_scan'               => 'snapshot_create'},
+        :snapshot_complete  => {'snapshot_create'           => 'scanning',
+                                'snapshot_delete'           => 'synchronizing'},
+        :snapshot_delete    => {'scanning'                  => 'snapshot_delete'},
+        :data               => {'snapshot_create'           => 'scanning',
+                                'scanning'                  => 'scanning',
+                                'snapshot_delete'           => 'snapshot_delete',
+                                'synchronizing'             => 'synchronizing',
+                                'finished'                  => 'finished'}
+      )
+    end
+  end
+
+  def before_scan
+    signal(:start_snapshot)
+  end
+
+  def call_snapshot_create
+    _log.info("Enter")
+
+    begin
+      context[:snapshot_mor] = nil
+
+      options[:snapshot] = :skipped
+      options[:use_existing_snapshot] = false
+      return unless create_snapshot
+      signal(:snapshot_complete)
+    rescue => err
+      _log.log_backtrace(err)
+      signal(:abort, err.message, "error")
+      return
+    rescue Timeout::Error
+      msg = case options[:snapshot]
+            when :smartProxy, :skipped then "Request to log snapshot user event with EMS timed out."
+            else "Request to create snapshot timed out"
+            end
+      _log.error(msg)
+      signal(:abort, msg, "error")
+    end
+  end
+
+  def call_snapshot_delete
+    _log.info("Enter")
```

Comment:
Do we want to include the method name in the log entry here?

Reply:
@djberg96 while I agree, I would do that in a separate PR. I am just refactoring here - trying to change the least amount possible and ensure everything still works.